### ISSUE TITLE
[Flang] Incorrect diagnostic of defined assignment

### ISSUE THREAD

**Issue by DanielCChen** [llvm/llvm-project#122490]
Consider the following code

```
module m

   type base
      contains
         procedure, pass(b) :: U_base
         generic :: assignment(=) => U_base
   end type

   contains

   subroutine U_base ( a, b )
      class(*), intent(out) :: a
      class(base), intent(in)   :: b

   end subroutine

end module

```

Flang currently issues an error as
```
./t.f:8:38: error: Defined assignment subroutine 'u_base' conflicts with intrinsic assignment
           generic :: assignment(=) => U_base
                                       ^^^^^^
./t.f:13:15: Declaration of 'u_base'
     subroutine U_base ( a, b )
                ^^^^^^
```

This seems incorrect. The compilers that I have tried (gfortran, ifort and XLF) all compiled it successfully.

**Comment by klausler** [comment]
Ok, good, thanks, I'm swamped with bugs at the moment.  But assign it to me if it turns out that you need to, and I'll handle it.

**Comment by klausler** [comment]
Assigning to you (Daniel) for bookkeeping purposes; please reassign it to me at your discretion if necessary.

**Comment by klausler** [comment]
Ping.  Do you want me to take over this bug?

**Comment by DanielCChen** [comment]
> Ping. Do you want me to take over this bug?

Sure. Sorry I haven't got the chance to look at it.

**Comment by klausler** [comment]
I no longer observe an error message on the test case above, using a compiler built from the current head of llvm-project/main.  Please close this bug if you can't reproduce it either.

**Comment by DanielCChen** [comment]
The code is passing now.
